luaui/Widgets/cmd_nano_range_check.lua

```diff
@@ -0,0 +1,277 @@
+local WIDGET_NAME = "Construction Turrets Range Check"
+local WIDGET_VERSION = "1.5"
+--[[
+### VERSIONS ###
+1.0 - initial release, basic
+1.1 - added more command types (reclaim, attack)
+1.2 - added support for queued commands to be flag_processed
+1.3 - fixed a range deviation caused by the game adding model radius
+1.4a - optimization, added LRU cache to reduce the number of calls to the engine
+1.4b - optimization, changed the flag_listening method to await a command instead of polling x'th frame
+1.4c - optimization, added a command limit to prevent the engine from ignoring commands
+1.4d - optimization, replaced loop with GiveOrderToUnitArray, renaiming and adding comments
+1.4e - changed distance calculation from 3D to 2D
+1.4f - removed a Spring.Echo() that was left from debugging.
+1.4g - refactored the code, added more comments, better readability.
+1.5 - fixed edge case that allows to skip the script the execution of the script by switching selection.
+1.5b - main branch integration. Removal of hello Echo.
+]]--
+
+function widget:GetInfo()
+    return {
+        name = WIDGET_NAME,
+        desc = "Stops construction turrets from being assigned to constructions out of reach.",
+        author = "Nehroz",
+        date = "2024.11.01", -- update date.
+        license = "GPL v3",
+        layer = 0,
+        version = WIDGET_VERSION
+    }
+end
+
+-- SECTION OOP
+-- SECTION LRU Cache class
+LRUCache = {}
+LRUCache.__index = LRUCache
+
+-- Constructor
+function LRUCache:new(max_size)
+    assert(type(max_size) == "number")
+    local cache = {
+        max_size = max_size or 10, -- Default max size to 10 if not specified
+        cache = {},                -- Key-Value store (uID -> value = radius)
+        order = {}                 -- To track the order of use (most recent at the end)
+    }
+    setmetatable(cache, LRUCache)
+    return cache
+end
+
+-- Get a value by uID
+function LRUCache:get(uID)
+    local value = self.cache[uID]
+    if value then
+        -- Move the accessed uID to the end to mark it as most recently used
+        self:moveToEnd(uID)
+        return value
+    else
+        return nil -- uID not found
+    end
+end
+
+-- Put a uID and value into the cache
+function LRUCache:put(uID, value)
+    if self.cache[uID] then
+        -- If uID already exists, just update and mark it as recently used (should never be the case)
+        self.cache[uID] = value
+        self:moveToEnd(uID)
+    else
+        -- Add new uID-value pair
+        if #self.order >= self.max_size then
+            -- Cache is full, remove the least recently used item
+            local lru = table.remove(self.order, 1)
+            self.cache[lru] = nil
+        end
+        table.insert(self.order, uID)
+        self.cache[uID] = value
+    end
+end
```

Comment:
This may be more legible as 

```lua
function LRUCache:put(uID, value)
    if self.cache[uID] then
        -- Update the value and mark it as recently used
        self.cache[uID] = value
        self:moveToEnd(uID)
        return
    end

    -- Check if the cache is full
    if #self.order >= self.max_size then
        -- Remove the least recently used item
        local lru = table.remove(self.order, 1)
        self.cache[lru] = nil
    end

    -- Insert the new uID and its value
    table.insert(self.order, uID)
    self.cache[uID] = value
end
```